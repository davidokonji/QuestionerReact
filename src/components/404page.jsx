import React, { Fragment } from 'react';
import Header from './Landingheader';

const NotFound = (props) => {
  const { text } = props;
  return (
    <Fragment>
      <Header
        page="landingpage"
        navClass="nav_header"
      />
      <div id="notfound">
        <div className="notfound">
          <div className="notfound-404">
            <h3>
            Oops!
              {
              text || ' Page '
            }
              not found

            </h3>
            <h1>
              <span>4</span>
              <span>0</span>
              <span>4</span>
            </h1>
          </div>
          <h2>
          we are sorry, page you requested was not found
          </h2>
          <a href="/" className="homeLink">home</a>
        </div>
      </div>
    </Fragment>
  );
};

export default NotFound;
