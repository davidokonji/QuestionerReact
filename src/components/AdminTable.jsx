import React from 'react';
import MeetupList from './MeetupList';

const AdminTable = ({ data, toggle }) => (

  // const deleteMeeteps = ( id) => {
  //     this.props.delete(id)
  // }
  <div className='col-11 mx-4 mx-md-0 mt-4 mt-md-0 col-md-9 shadow p-0 table-responsive-sm'>
    <table className='table table-hover p-0'>
      <thead className='row mx-0'>
        <div className='col-10 bg-light p-3'>
            Available Meetup(s)
        </div>
        <div className='col-2 bg-light p-3'>
            Operation
        </div>
      </thead>
      <tbody className='row p-2 col-12 ml-0 '>
        {
          data.map(meetup => (
            <MeetupList
              id={meetup.id}
              title={meetup.title}
              location={meetup.location}
              tags={meetup.tags}
              happeningOn={meetup.happeningOn}
              images={meetup.images}
              key={meetup.id}
              // delete={delteMeetups}
            //   deleteDialog={toggle}
            />
          ))
      }
      </tbody>
    </table>
  </div>
);

export default AdminTable;
