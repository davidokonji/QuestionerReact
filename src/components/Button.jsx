import React from 'react';

const Button = (props) => {
  const { text, event, styles } = props;
  return (
    <button type="button" onClick={event} className={styles}>
      {text}
    </button>
  );
};

export default Button;
