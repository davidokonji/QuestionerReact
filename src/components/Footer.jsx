import React from 'react';
import '../style/footer.scss';

const Footer = () => {
  const date = new Date().getFullYear();
  return (
    <React.Fragment>
      <footer>
      &copy;
        {date}

      </footer>
    </React.Fragment>
  );
};

export default Footer;
