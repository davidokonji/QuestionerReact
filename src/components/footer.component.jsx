import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Footer = () => (
  <footer className="footer">
    <div className="footerBottom">
      <p>
       Created with
        {<FontAwesomeIcon icon="heart" size="sm" color="red" style={{ paddingLeft: '1rem', paddingRight: '1rem' }} />}
       by Okonji Chukwunonso
      </p>
    </div>
  </footer>
);

export default Footer;
