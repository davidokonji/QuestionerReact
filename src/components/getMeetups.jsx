import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { getMeetups } from '../actions';
import Card from './common/Card';

export class GetMeetups extends Component {
  async componentDidMount() {
    const { allMeetups } = this.props;
    const token = localStorage.getItem('token');
    await allMeetups(token);
  }

  render() {
    const { meetups: { data = [] } } = this.props;
    return (
      <div className='container-fluid mb-5 mx-md-5 mx-0'>
        <div className='row mb-5 my-5'>
          {
          data.map(meetup => (
            <Card
              id={meetup.id}
              title={meetup.title}
              location={meetup.location}
              tags={meetup.tags}
              happeningOn={meetup.happeningOn}
              images={meetup.images}
              key={meetup.id}
            />
          ))
      }
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = {
  allMeetups: getMeetups,
};
export const mapStateToProps = state => ({
  meetups: state.meetups,
});

export default connect(mapStateToProps, mapDispatchToProps)(GetMeetups);
