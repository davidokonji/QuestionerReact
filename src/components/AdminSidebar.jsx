import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import { WithContext as ReactTags } from 'react-tag-input';
import {
  Form,
  FormGroup,
  Label,
  Input,
  FormText,
} from 'reactstrap';
import Button from './Button';
import { addMeetup } from '../actions';
import Modal from './common/modal';

const KeyCodes = {
  comma: 188,
  enter: 13,
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];
export class AdminSidebar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tagList: [
      ],
      suggestions: [
        { id: 'technology', text: 'technology' },
        { id: 'social', text: 'social' },
      ],
      topic: '',
      location: '',
      happeningOn: '',
      images: '',
      imagePreview: ''
    };
  }

  handleOnChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  imageHandler = ({ target: { files } }) => {
    this.setState({
      images: files[0],
      imagePreview: URL.createObjectURL(files[0]),
    });
  }


  handleSubmit = (e) => {
    e.preventDefault();
    const { createMeetup } = this.props;
    const {
      topic,
      location,
      happeningOn,
      tagList,
      images
    } = this.state;
    const filteredTags = tagList.map(tag => tag.text);
    const tags = filteredTags.join();
    const data = {
      topic,
      location,
      happeningOn,
      tags,
      images
    };

    createMeetup(data);
  }

  handleAddition = (tag) => {
    this.setState(state => ({ tagList: [...state.tagList, tag] }));
  }

  handleDrag = (tag, currPos, newPos) => {
    const { tagList } = this.state;
    const setTags = new Set([...tagList]);
    const newTags = setTags.slice();

    newTags.splice(currPos, 1);
    newTags.splice(newPos, 0, tag);

    this.setState({ tagList: newTags });
  }

  handleDelete = (i) => {
    const { tagList } = this.state;
    this.setState({
      tagList: tagList.filter((tag, index) => index !== i),
    });
  }

  render() {
    const { toggle, open } = this.props;
    const {
      tagList,
      suggestions,
      imagePreview,
      topic
    } = this.state;
    const placeholder = 'Add meetup Tags';
    return (
      <Fragment>
        <div className='col-12 col-md-3'>
          <div className='p-5 shadow-sm'>
            <Button styles='button p-3' text='Create Meetup' event={toggle} />
          </div>
        </div>
        <Modal toggle={toggle} open={open} className='mt-5'>
          <Form method='post' onSubmit={this.handleSubmit} encType='multipart/form-data'>
            {
              imagePreview && (
                <div>
                  <img src={imagePreview} alt={topic} style={{ height: '30rem' }} />
                </div>
              )
            }
            <FormGroup>
              <Input
                type='file'
                name='image'
                max={1}
                accept='image/*'
                onChange={this.imageHandler}
              />
              <FormText color='muted'>
              Add meetup images
              </FormText>
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='topic'
                placeholder='Add Topic'
                onChange={this.handleOnChange}
                className='px-3'
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='text'
                name='location'
                placeholder='Add Location'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <Input
                type='date'
                name='happeningOn'
                placeholder='Add Date'
                onChange={this.handleOnChange}
              />
            </FormGroup>
            <FormGroup>
              <div id='app'>
                <ReactTags
                  tags={tagList}
                  suggestions={suggestions}
                  handleDelete={this.handleDelete}
                  handleAddition={this.handleAddition}
                  handleDrag={this.handleDrag}
                  delimiters={delimiters}
                  placeholder={placeholder}
                  inline={false}
                />
              </div>
            </FormGroup>
            <button type='submit' className='btn' onClick={toggle}>Create Meetup</button>
          </Form>
        </Modal>
      </Fragment>
    );
  }
}
const mapDispatchToProps = {
  createMeetup: addMeetup
};


export default connect(null, mapDispatchToProps)(AdminSidebar);
