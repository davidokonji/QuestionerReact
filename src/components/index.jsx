import React from 'react';
import Navbar from './Navbar';
import Slider from './slider';
import Footer from './Footer';

const App = () => {
  return (
    <React.Fragment>
      <Navbar />
      <Slider />
      <Footer />
    </React.Fragment>
  );
};

export default App;
