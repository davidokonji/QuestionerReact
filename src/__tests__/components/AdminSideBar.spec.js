import React from 'react';
import { shallow } from 'enzyme';
import { AdminSidebar } from '../../components/AdminSidebar';

describe('Admin sidebar component', () => {
  it('should render component correctly', () => {
    const toggle = jest.fn();
    const open = false;
    const wrapper = shallow(<AdminSidebar toggle={toggle} open={open} />);

    expect(wrapper).toMatchSnapshot();
  });
});
