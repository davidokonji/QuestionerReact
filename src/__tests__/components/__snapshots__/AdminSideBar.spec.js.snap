// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin sidebar component should render component correctly 1`] = `
<Fragment>
  <div
    className="col-12 col-md-3"
  >
    <div
      className="p-5 shadow-sm"
    >
      <Button
        event={[MockFunction]}
        styles="button p-3"
        text="Create Meetup"
      />
    </div>
  </div>
  <ModalComponent
    className="mt-5"
    open={false}
    toggle={[MockFunction]}
  >
    <Form
      encType="multipart/form-data"
      method="post"
      onSubmit={[Function]}
      tag="form"
    >
      <FormGroup
        tag="div"
      >
        <Input
          accept="image/*"
          max={1}
          name="image"
          onChange={[Function]}
          type="file"
        />
        <FormText
          color="muted"
          tag="small"
        >
          Add meetup images
        </FormText>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Input
          className="px-3"
          name="topic"
          onChange={[Function]}
          placeholder="Add Topic"
          type="text"
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Input
          name="location"
          onChange={[Function]}
          placeholder="Add Location"
          type="text"
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Input
          name="happeningOn"
          onChange={[Function]}
          placeholder="Add Date"
          type="date"
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          id="app"
        >
          <DragDropContext(ReactTags)
            delimiters={
              Array [
                188,
                13,
              ]
            }
            handleAddition={[Function]}
            handleDelete={[Function]}
            handleDrag={[Function]}
            inline={false}
            placeholder="Add meetup Tags"
            suggestions={
              Array [
                Object {
                  "id": "technology",
                  "text": "technology",
                },
                Object {
                  "id": "social",
                  "text": "social",
                },
              ]
            }
            tags={Array []}
          />
        </div>
      </FormGroup>
      <button
        className="btn"
        onClick={[MockFunction]}
        type="submit"
      >
        Create Meetup
      </button>
    </Form>
  </ModalComponent>
</Fragment>
`;
